htdocs/product/dynamic_price/class/price_parser.class.php

```diff
@@ -131,6 +131,15 @@ public function translatedError()
 	public function parseExpression($product, $expression, $values)
 	{
 		global $user;
+
+		//Check if empty
+		$expression = trim($expression);
+		if (empty($expression))
+		{
+			$this->error = array(20, null);
```

Comment:
Price Parser uses error var for passing a numeric error and extra values, it was always this way https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/product/dynamic_price/class/price_parser.class.php